portfolio/src/main/webapp/script.js

```diff
@@ -86,9 +86,78 @@ function initMap() {
   map.mapTypes.set('styled_map', styledMapType);
   map.setMapTypeId('styled_map');
 
+  map.addListener('click', (event) => {
+    createVisitorMarker(map, event.latLng.lat(), event.latLng.lng());
+  });
+
   addMyMapMarkers(map);
+  addVisitorMarkers(map);
+}
+
+/** Fetches markers from the backend and adds them to the map. */
+function addVisitorMarkers(map) {
+  fetch('/markers').then(response => response.json()).then((markers) => {
+    for(marker of markers){
+      const visitorMarker = new google.maps.Marker({
+        title: 'A visitor is from here!',
+        position: {lat: marker.lat, lng: marker.lng},
+        map: map
+      });
+    }
+  });
+}
+
+/** Creates a marker where visitors can submit. */
+function createVisitorMarker(map, lat, lng) {
```

Comment:
There is too much happening in this method and it was hard to read. 

Instead:
```
a() {
  ...
  b()
}
 b() {
  ...
 c()
}

c() {
  ...
}

a()
```

It is easier to read if it was
```
a() {
  ...
}

b() {
  ...
}

c() {
  ...
}

a()
b()
c()
```

It is not always as simple as in the example above (especially if there are the methods are attached to listeners) but please attempt this with your best judgement and I'll then re-review.

Reply:
Good enough! Thank you! Better to read now!